recipes/automated-testing.md

```diff
@@ -111,13 +127,16 @@ time decoupling logic testing and endpoint testing.
 - To verify the API endpoint performs authentication and authorization.
 - To verify user permissions for that endpoint.
 - To verify that invalid input is correctly handled.
+- To verify the basic business logic is handled correctly, both in expected success and failure cases.
+- To verify infrastructure related side-effects, e.g. database changes or calls to third party services.
 
 #### When **not** to use
-- For testing of specific function logic. We should use unit tests for those.
+- For extensive testing of business logic permutations beyond fundamental scenarios. Integration tests contain more overhead to write compared to unit tests and can easily lead to a combinatorial explosion. Instead, unit tests should be used for thorough coverage of these permutations.
```

Comment:
What would combinatorial explosion mean, and how would it look like?

Reply:
It is the fast growth in the number of combinations that need to be tested when multiple business rules are involved. At least that's the term I've seen sometimes used in the context of writing tests 😄 It's especially important in integration tests since they have a higher chance of invoking multiple different business rules at the same time. Didn't explain the term because I guessed it could be googled. What do you think? Should a different term be used?

Reply:
It falls in line with what was previously written that unit tests should be preferred way of testing each unit logic instead of testing multiple units and trying to add enough permutations/combinations to cover all use cases.